### PR TITLE
[ext-docs] Remove "What are extensions?"

### LINKED ISSUE #4322: What are Extensions? (Overview Refresh)
Rewrite the general extensions overview so that it can work better with Extensions 101, Development Basics, and Architecture Overview.

### PR DESCRIPTION
**Pull request by AmySteam**
Fixes #4322 

Extensions 101 and Development basics all the information presented in [What are extensions?](https://developer.chrome.com/docs/extensions/mv3/overview/). Therefore, this doc has become redundant. 

Changes proposed in this pull request:

- Consolidate any missing information into Extensions 101.
- Fix all the links that pointed to this doc. 
- Add a redirect from overview to architecture-overview.

**Note**

This PR has been reviewed for technical accuracy on [PR#4533](https://github.com/GoogleChrome/developer.chrome.com/pull/4533)

### CODE REVIEW

File: site/en/docs/extensions/mv3/getstarted/extensions-101/index.md [hunk 1]
```diff
@@ -114,10 +122,25 @@ See [Extension quality guidelines][doc-single-purpose] for additional details.
...
+policies][doc-cws-policy].
 
 See [Publish in the Chrome Web Store][doc-cws-publish] to learn how to distribute your extension.
 
+
+
+{% Details %} 
+{% DetailsSummary %}
+
+**💡 What if I only want to distribute the extension within my organization?**
```
Comment (jpmedley) [inline site/en/docs/extensions/mv3/getstarted/extensions-101/index.md:134]:
```suggestion
**What if I only want to distribute the extension within my organization?**
```

Comment (jpmedley) [reply site/en/docs/extensions/mv3/getstarted/extensions-101/index.md:134]: I'm not crazy about emojis in formal writing. I also suspect that if emojis had been invented when Strunk and White was written, they would have mentioned them under 'omit needless words.'

File: site/en/docs/extensions/mv3/getstarted/extensions-101/index.md [hunk 2]
```diff
@@ -114,10 +122,25 @@ See [Extension quality guidelines][doc-single-purpose] for additional details.
...
+
+{% endDetailsSummary %}
+
+Some organizations use [enterprise policies][doc-cws-enterprise] to install extensions on their user's devices. These
+extensions may either be fetched from the Chrome Web Store or hosted on the organization's web
+servers.
```
Comment (jpmedley) [inline site/en/docs/extensions/mv3/getstarted/extensions-101/index.md:140]:
```suggestion
Some organizations use enterprise policies to install extensions on their user's devices. These
extensions may either be fetched from the Chrome Web Store or hosted on the organization's web
servers.
Read about both in [Enterprise publishing options][doc-cws-enterprise].
```

File: site/en/docs/webstore/about_webstore/index.md [hunk 3]
```diff
@@ -21,13 +21,13 @@ build Chrome-specific features into it.
 
 ## How do I start?
 
-1.  Learn more about [Chrome Extensions][extensions]
+1.  Learn more about [Chrome Extension development][getstarted]
```
Comment (jpmedley) [inline site/en/docs/webstore/about_webstore/index.md:24]: Per our handbook, please reserve numbered items for procedures. Make this into bullets.

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: site/en/docs/webstore/about_webstore/index.md (hunk 3 above)
Comment (AmySteam) [reply site/en/docs/webstore/about_webstore/index.md:24]: I'm going to do this on a separate PR 👍

File: site/en/docs/extensions/mv3/getstarted/extensions-101/index.md (hunk 1 above)
Comment (AmySteam) [reply site/en/docs/extensions/mv3/getstarted/extensions-101/index.md:134]: I like it that it draws attention to a question that developers might be curious about. I have used emojis in the getting started guides as a friendly gesture for new developers. I'd like to keep them for now, and we can meet with the team to present the pros and cons of this approach. 

Let's merge this as is for now, I don't consider it a blocker.

**Comment by AmySteam** [comment]
@jpmedley Thanks for your review. I applied most of your suggestions; I'd like to address the others on a separate PR.
IMO this is ready to be merged.

**Comment by AmySteam** [comment]
@jpmedley This is ready to merge

**Review by jpmedley — APPROVED** [review]

**Pull request merged**